src/plugins/tool_extrude.py

```diff
@@ -109,21 +129,33 @@ def on_drag(self, target):
 
         if ty != 0 or tx != 0 or tz != 0:
             self._mouse = (target.mouse_x, target.mouse_y)
-        if tx != 0 and self.xdir:
-            self.ydir = False
-            self.zdir = False
-            self.pastoffset += tx
-            self.drawstamp(target, self.pastoffset, 0, 0)
-        if ty != 0 and self.ydir:
-            self.xdir = False
-            self.zdir = False
-            self.pastoffset += ty
-            self.drawstamp(target, 0, self.pastoffset, 0)
-        if tz != 0 and self.zdir:
-            self.xdir = False
-            self.ydir = False
-            self.pastoffset += tz
-            self.drawstamp(target, 0, 0, self.pastoffset)
+
+        if self.fixeddirection:
+            delt = tx + ty + tz
+            if delt != 0:
+                self.pastoffset += delt
+                if self.xdir:
+                    self.drawstamp(target, self.pastoffset, 0, 0)
+                if self.ydir:
+                    self.drawstamp(target, 0, self.pastoffset, 0)
+                if self.zdir:
+                    self.drawstamp(target, 0, 0, self.pastoffset)
+        else:
+            if tx != 0 and self.xdir:
+                self.ydir = False
+                self.zdir = False
+                self.pastoffset += tx
+                self.drawstamp(target, self.pastoffset, 0, 0)
+            if ty != 0 and self.ydir:
+                self.xdir = False
+                self.zdir = False
+                self.pastoffset += ty
+                self.drawstamp(target, 0, self.pastoffset, 0)
+            if tz != 0 and self.zdir:
+                self.xdir = False
+                self.ydir = False
+                self.pastoffset += tz
+                self.drawstamp(target, 0, 0, self.pastoffset)
```

Comment:
also prioritizing x over y over z is not ideal here, the one with the highest abs of the non sign t[xyz] value should be prioritized here